api/proto/teleport/inventory/v1/inventory_service.proto

```diff
@@ -0,0 +1,100 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package teleport.inventory.v1;
+
+import "teleport/legacy/types/types.proto";
+import "teleport/machineid/v1/bot_instance.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/inventory/v1;inventoryv1";
+
+// InventoryService provides methods to list instances from the inventory cache. This service does not fallback to the backend if the cache is unhealthy.
+service InventoryService {
+  // ListUnifiedInstances returns a page of teleport instances and bot_instances. This API will refuse any requests when the cache is unhealthy or not yet
+  // fully initialized.
+  rpc ListUnifiedInstances(ListUnifiedInstancesRequest) returns (ListUnifiedInstancesResponse);
+}
+
+// UnifiedInstanceSort specifies the sort mode for listing unified instances.
+enum UnifiedInstanceSort {
+  UNIFIED_INSTANCE_SORT_UNSPECIFIED = 0;
+  // UNIFIED_INSTANCE_SORT_NAME sorts by display name (hostname for instances, bot name for bot instances).
+  UNIFIED_INSTANCE_SORT_NAME = 1;
+  // UNIFIED_INSTANCE_SORT_TYPE sorts by instance type (instance vs bot_instance).
+  UNIFIED_INSTANCE_SORT_TYPE = 2;
+  // UNIFIED_INSTANCE_SORT_VERSION sorts by version.
+  UNIFIED_INSTANCE_SORT_VERSION = 3;
+}
+
+// SortOrder specifies the sort order for listing unified instances.
+enum SortOrder {
+  SORT_ORDER_UNSPECIFIED = 0;
+  // SORT_ORDER_ASCENDING sorts in ascending order.
+  SORT_ORDER_ASCENDING = 1;
+  // SORT_ORDER_DESCENDING sorts in descending order.
+  SORT_ORDER_DESCENDING = 2;
+}
+
+// ListUnifiedInstancesRequest is the request for listing instances.
+message ListUnifiedInstancesRequest {
+  // page_size is the size of the page to return.
+  int32 page_size = 1;
+  // page_token is the next_page_token value returned from a previous ListUnifiedInstances request, if any.
+  string page_token = 2;
+  // filter specifies optional search criteria to limit which instances should be returned.
+  ListUnifiedInstancesFilter filter = 3;
+  // sort specifies the sort mode for the results. Defaults to UNIFIED_INSTANCE_SORT_NAME.
+  UnifiedInstanceSort sort = 4;
```

Comment:
Do we need/want to enforce some specific sort mode when filtering in specific ways? if we're choosing a single `instance_types` we might want to prefer UNIFIED_INSTANCE_SORT_TYPE over UNIFIED_INSTANCE_SORT_NAME for example.